slack-user.c

```diff
@@ -148,13 +148,27 @@ void slack_user_retrieve(SlackAccount *sa, const char *uid, SlackUserCallback *c
 	slack_api_call(sa, user_retrieve_cb, lookup, "users.info", "user", uid, NULL);
 }
 
+static void self_presence_cb(SlackAccount *sa, gpointer data, json_value *json, const char *error) {
+	if (json_get_prop_boolean(json, "manual_away", TRUE))
+		return;
+
+	if (sa->ping_timer) {
+		purple_timeout_remove(sa->ping_timer);
+		sa->ping_timer = 0;
+	}
+	sa->login_step = 0;
+	slack_login_step(sa);
```

Comment:
Could you just use `slack_rtm_connect(sa)` instead, so that the buddy list and conversations list aren't redownloaded?

Reply:
Just switched, but it required more changes than I would've liked and causes `g_log: purple_websocket_send: assertion 'ws->connected && !(ws->closed & PURPLE_INPUT_WRITE)' failed` on occasion, which I'm not sure whether will end up causing issues or not. Pushed anyway.